FinancialTracker/ViewControllers/RemindersTableViewController.swift

```diff
@@ -116,16 +116,19 @@ extension RemindersTableViewController: UITableViewDataSource {
     func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
         if editingStyle == .delete {
             let reminder = transfers[indexPath.section].value[indexPath.row]
-            authManager?.deleteReminderFromCurrentUser(reminder: reminder, completionHandler: { authError, _ in
-                if let alert = UIAlertController.create(basedOn: authError) {
-                    self.present(alert, animated: true)
+            authManager?.deleteReminderFromCurrentUser(reminder: reminder, completionHandler: { authError, success in
+                guard success else {
+                    let alertTitle = authError?.title ?? "Unknown Error"
+                    let alertMessage = authError?.message ?? "This error should not appear."
+                    
+                    self.present(UIAlertController.create(title: alertTitle, message: alertMessage), animated: true)
                     return
-                } else {
-                    let transfers = self.transfers[indexPath.section].value
-                    let key = self.transfers[indexPath.section].key
-                    if let index = transfers.firstIndex(of: reminder) {
-                        self.transfers[key]?.remove(at: index)
-                    }
+                }
+                
+                let transfers = self.transfers[indexPath.section].value
+                let key = self.transfers[indexPath.section].key
+                if let index = transfers.firstIndex(of: reminder) {
+                    self.transfers[key]?.remove(at: index)
```

Comment:
should there be an `else` for this `if let` statement?

Reply:
Not needed it would not be to delete an index that doesn't exist it automatically passes the correct one when you swipe to delete.

Reply:
nevertheless, better put in an `else` with an `assertionFailure` for just in case